src/tasks/MonoTargetsTasks/NetTraceToMibcConverterTask/NetTraceToMibcConverter.cs

```diff
@@ -0,0 +1,115 @@
+using System.Text.Json.Serialization;
+
+
+public class NetTraceToMibcConverter : ToolTask
+{
+    /// <summary>
+    /// Entries for assemblies referenced in a .nettrace file. Important when you run traces against an executable on a different machine / device
+    /// </summary>
+    [Required]
+    public ITaskItem[] Assemblies { get; set; } = Array.Empty<ITaskItem>();
+
+    /// <summary>
+    /// NetTrace file to use when invoking dotnet-pgo for
+    /// </summary>
+    [Required]
+    public string NetTracePath { get; set; } = "";
+
+    /// <summary>
+    /// Directory where the mibc file will be placed
+    /// </summary>
+    [NotNull]
+    [Required]
+    public string? OutputDir { get; set; }
+
+    /// <summary>
+    /// The path to the mibc file generated from the converter.
+    /// </summary>
+    [Output]
+    public string? MibcProfilePath { get; set; }
+
+    protected override string ToolName { get; } = "NetTraceToMibcConverter";
+
+    protected override string GenerateFullPathToTool()
+    {
+        return ToolPath;
+    }
+
+    protected override bool ValidateParameters()
+    {
+        if (string.IsNullOrEmpty(ToolPath))
+        {
+            Log.LogError($"{nameof(ToolPath)}='{ToolPath}' must be specified.");
+            return false;
+        }
+
+        if (string.IsNullOrEmpty(ToolExe))
+        {
+            Log.LogError($"{nameof(ToolExe)}='{ToolExe}' must be specified.");
+            return false;
+        }
+
+        string mibcConverterBinaryPath = Path.Combine(ToolPath, ToolExe);
+
+        if (!File.Exists(mibcConverterBinaryPath))
+        {
+            Log.LogError($"{nameof(mibcConverterBinaryPath)}='{mibcConverterBinaryPath}' doesn't exist.");
+            return false;
+        }
+
+        if (Assemblies.Length == 0)
+        {
+            Log.LogError($"'{nameof(Assemblies)}' is required.");
+            return false;
+        }
+
+        if (!File.Exists(NetTracePath))
+        {
+            Log.LogError($"{nameof(NetTracePath)}='{NetTracePath}' doesn't exist");
+            return false;
+        }
+
+        foreach (var asmItem in Assemblies)
+        {
+            string? fullPath = asmItem.GetMetadata("FullPath");
+            if (!File.Exists(fullPath))
+                throw new LogAsErrorException($"Could not find {fullPath} to AOT");
+        }
+
+        return !Log.HasLoggedErrors;
+    }
+
+    protected override string GenerateCommandLineCommands()
+    {
+        var outputMibcPath = Path.Combine(OutputDir, Path.ChangeExtension(Path.GetFileName(NetTracePath), ".mibc"));
+
+        StringBuilder mibcConverterArgsStr = new StringBuilder(string.Empty);
+        mibcConverterArgsStr.Append($"create-mibc");
+        mibcConverterArgsStr.Append($" --trace {NetTracePath} ");
+
+        foreach (var refAsmItem in Assemblies)
+        {
+            string? fullPath = refAsmItem.GetMetadata("FullPath");
+            mibcConverterArgsStr.Append($" --reference \"{fullPath}\" ");
```

Comment:
should we throw if fullPath is empty?

Reply:
Can an Item not have a FullPath?
Maybe it can just be `string fullPath`

Reply:
`FullPath` is a reserved metadata, and will always be non-null. Even for non-path-like item values.

Reply:
We throw if fullPath is empty in `ValidateParameters`